### ISSUE TITLE
Reliable detecting the Zope version

### ISSUE THREAD

**Issue by d-maurer** [zopefoundation/Zope#411]
Add-ons interacting with the ZMI need to provide different templates for Zope 2 (old style ZMI) and Zope 4 (bootstrap based ZMI). Therefore, there is a need to reliable distinguish those versions -- potentially via a mechanism analogous to Python's `sys.version_info`.

**Comment by d-maurer** [comment]
`Zope 2` used to have a function `getZopeVersion` in `App.version_txt` for this purpose. This function seems to have been lost. Can it be restored?

**Comment by icemac** [comment]
Nowadays I'd use `pkg_resources` to get the version number like this:
```python
import pkg_resources
zope_version = pkg_resources.get_distribution('Zope').version
```

Could it be a solution in your use case, too?

**Comment by d-maurer** [comment]
> ... `pkg_resources.get_distribution('Zope').version` ... Could it be a solution in your use case, too?

In fact, Zope2's `getZopeVersion` already used this approach internally. But, it employs an impressive regular expression to extract and repackage the version information -- code I would not like to reproduce.

There is another disadvantage with your proposed solution for my current use case, i.e. making extensions working fine in both "Zope2" and "Zope 4+". For "Zope2", I must look at the distribution "Zope2"; for "Zope 4+" at the distribution "Zope". In a typical "Zope 4" setup, there is also a "Zope2" distribution but apparently there is no strict relationship between the versions of "Zope" and "Zope2" (at the moment "4.0b7" and "4.0b1", respectively). This lets it seem doubtful to always look at the "Zope2" distribution version to determine the version of "Zope 4+".

In short: determining the version via `pkg_resources` would be possible but nasty. Currently, I use:
```
try:
  from App.version_txt import getZopeVersion
  zope_major_version = getZopeVersion()[0]
except ImportError: zope_major_version = 4 # workaround for Zope 4 beta bug
```

**Comment by icemac** [comment]
@d-maurer I get your point. I added this issue to the list of open issues for the final release. A pull request is highly welcome.

Note to the one who will fix this issue: `getZopeVersion()` was removed in eb3b5a12d6378fb7cdfce09e4beb9bafabb644ea.
